istioctl/cmd/analyze.go

```diff
@@ -188,7 +189,19 @@ func Analyze() *cobra.Command {
 				if err != nil {
 					return err
 				}
-				sa.AddRunningKubeSource(k)
+				if allNamespaces {
+					sa.AddRunningKubeSource(k)
+					revisions, _ := tag.ListRevisions(context.TODO(), k.Kube())
```

Comment:
injection revision and config revision are orthogonal. A namespace can be configured with rev1 and have CRs for rev2, rev3, etc.